src/main/java/org/terasology/launcher/TerasologyLauncher.java

```diff
@@ -16,111 +16,86 @@
-
-    private Pane splashLayout;
-    private ProgressBar loadProgress;
-    private Label progressText;
-    private Stage mainStage;
-    private HostServices hostServices;
+    private final LauncherInitTask initTask = new LauncherInitTask();
+    private LauncherConfiguration initialConfig;
+    private MainController mainController;
 
     public static void main(String[] args) {
         launch(args);
     }
 
     @Override
-    public void init() throws Exception {
-        ImageView splash = new ImageView(BundleUtils.getFxImage("splash"));
-        loadProgress = new ProgressBar();
-        loadProgress.setPrefWidth(SPLASH_WIDTH);
-        progressText = new Label();
-        splashLayout = new VBox();
-        splashLayout.getChildren().addAll(splash, loadProgress, progressText);
-        progressText.setAlignment(Pos.CENTER);
-        splashLayout.getStylesheets().add(BundleUtils.getStylesheet("css_splash"));
-        splashLayout.setEffect(new DropShadow());
-        hostServices = getHostServices();
+    public void init() {
+        initTask.setOnSucceeded(event -> {
+            if (mainController != null) {
+                mainController.update(initTask.getValue());
+            } else {
+                initialConfig = initTask.getValue();
+            }
+        });
+        initTask.setOnFailed(event -> openCrashReporterAndExit((Exception) initTask.getException()));
+        new Thread(initTask).start();
     }
 
     @Override
-    public void start(final Stage initialStage) {
+    public void start(Stage mainStage) throws IOException {
         logger.info("TerasologyLauncher is starting");
-        logSystemInformation();
 
+        logSystemInformation();
         initProxy();
         initLanguage();
 
-        final Task<LauncherConfiguration> launcherInitTask = new LauncherInitTask(initialStage);
-
-        showSplashStage(initialStage, launcherInitTask);
-        Thread initThread = new Thread(launcherInitTask);
-
-        launcherInitTask.setOnSucceeded(new EventHandler<WorkerStateEvent>() {
-            @Override
-            public void handle(final WorkerStateEvent workerStateEvent) {
-                try {
-                    LauncherConfiguration config = launcherInitTask.getValue();
-                    if (config == null) {
-                        throw new LauncherStartFailedException("Launcher configuration was `null`.");
-                    } else if (config instanceof NullLauncherConfiguration) {
-                        logger.info("Closing the launcher ... (empty configuration, probably due to update)");
-                        Platform.exit();
-                    } else {
-                        showMainStage(config);
-                    }
-                } catch (IOException | LauncherStartFailedException e) {
-                    openCrashReporterAndExit(e);
-                }
+        FXMLLoader fxmlLoader = BundleUtils.getFXMLLoader("main_view");
```

Comment:
Please keep the `showMainStage` method. IMO it is helpful if all statements of a method describe something on (somewhat) the same semantic level. For instance, `start()` can be summarized as 

- log system information
- init proxy settings
- init languages
- show main stage